src/main/scala/ai/datahunters/md/reader/SolrHashReader.scala

```diff
@@ -0,0 +1,81 @@
+package ai.datahunters.md.reader
+
+import java.util
+
+import ai.datahunters.md.config.writer.SolrWriterConfig
+import ai.datahunters.md.util.SolrClientBuilder
+import org.apache.solr.client.solrj.impl.CloudSolrClient
+import org.apache.solr.common.SolrDocument
+import org.apache.solr.common.params.MapSolrParams
+import org.apache.spark.sql.types.{DataTypes, StructField, StructType}
+import org.apache.spark.sql.{DataFrame, Row, SparkSession}
+
+import scala.collection.JavaConversions._
+
+/**
+  * Read  id and hashes to DataFrame mainly for processing reduction
+  *
+  * @param sparkSession
+  * @param config
+  */
+case class SolrHashReader(sparkSession: SparkSession,
+                          config: SolrWriterConfig) extends PipelineSource {
+
+  import ai.datahunters.md.reader.SolrHashReader._
+
+  private val documentToStrings: SolrDocument => Seq[String] =
+    (document: SolrDocument) => HashSeq.map(s => getDocumentFieldValueOrEmptyString(document, s))
+
+
+  override def load(): DataFrame = {
+    val client = SolrClientBuilder().setZKServers(config.zkServers)
+      .setZKSolrChroot(config.zkSolrZNode)
+      .setDefaultCollection(config.collection)
+      .build()
+
+    val documents = getSolrDocumentList(client)
+
+    val idHashValueSeq = documents.map((d:SolrDocument) =>
+      Row.fromSeq(documentToStrings(d)))
+
+    val dataRdd = sparkSession.sparkContext.parallelize(idHashValueSeq)
+    sparkSession.createDataFrame(dataRdd, Schema)
```

Comment:
@dawidsau Building DF in driver is risky. We have to get all those data from Solr (I know it's only string but it's could be still huge data) to memory and then build DF. We should probably build new type of Spark DataSource. I remember that there is one existing implemented by Lucidworks but it's huge and I don't want to have this library in our dependencies. I wonder if there is another way for doing it. I have some ideas but maybe we should do this in next iteration and now mark this feature as experimental (potentially causing OOM and requiring more memory on driver side) and include in release. What do you think?

Reply:
Totally agree with you. I have added relevant comments in Readme.